From 2bb4c72468b6247275ad99f279fe718f5a7f134e Mon Sep 17 00:00:00 2001
From: Alex Wilson <alex.wilson@joyent.com>
Date: Mon, 24 Apr 2017 14:16:31 -0700
Subject: [PATCH] joyent/node-cueball#109 Agent "health check ok" log msg needs
 domain name of pool Reviewed by: David Pacheco <dap@joyent.com>

---
 lib/agent.js | 28 ++++++++++++++++++++++++----
 1 file changed, 24 insertions(+), 4 deletions(-)

diff --git a/lib/agent.js b/lib/agent.js
index 74e1d07..d5362a5 100644
--- a/lib/agent.js
+++ b/lib/agent.js
@@ -335,7 +335,13 @@ CueBallAgent.prototype.addRequest = function (req, optionsOrHost, port) {
 };
 
 CueBallAgent.prototype.checkSocket = function (host, handle, socket) {
-	var self = this;
+	var t1 = new Date();
+	var log = handle.ch_slot.makeChildLogger({
+		component: 'CueBallAgent',
+		domain: host,
+		localPort: socket.localPort,
+		path: this.cba_ping
+	});
 	var agent = new PingAgent({
 		protocol: this.protocol,
 		socket: socket,
@@ -358,16 +364,30 @@ CueBallAgent.prototype.checkSocket = function (host, handle, socket) {
 		});
 		res.on('end', function () {
 			if (res.statusCode >= 500 && res.statusCode < 600) {
-				self.log.warn('got a 500, closing');
+				var t2 = new Date();
+				log.warn({
+					statusCode: res.statusCode,
+					latency: (t2.getTime() - t1.getTime())
+				}, 'got a 5xx code, closing');
 				handle.close();
 			} else {
-				self.log.debug('health check ok, releasing');
+				log.debug({
+					statusCode: res.statusCode
+				}, 'health check ok, releasing');
 				handle.release();
 			}
 		});
 	});
 	req.once('error', function (e) {
-		self.log.warn(e, 'check failed');
+		var t2 = new Date();
+		log.warn({
+			err: {
+				name: e.name,
+				stack: e.stack,
+				message: e.message
+			},
+			latency: (t2.getTime() - t1.getTime())
+		}, 'check failed');
 		handle.close();
 	});
 	req.end();
-- 
2.21.0

